src/Agent.Worker/StepsRunner.cs

```diff
@@ -33,6 +34,10 @@ public interface IStepsRunner : IAgentService
 
     public sealed class StepsRunner : AgentService, IStepsRunner
     {
+        private IJobServerQueue _jobServerQueue;
+
+        private IJobServerQueue JobServerQueue => _jobServerQueue ??= HostContext.GetService<IJobServerQueue>();
```

Comment:
HostContext can be null here because is initialized by the Initialize method which theoretically may not be called. May be it's better to implement JobServerQueue like HostContext and override Initialize method here with JobServerQueue initialization